pkg/controller/images/images.go

```diff
@@ -139,35 +140,22 @@ func createImageForSelf(req router.Request, app *v1.AppInstance) (*v1.ImageInsta
 	return image, req.Client.Create(req.Ctx, image)
 }
 
-func removeTag(req router.Request, app *v1.AppInstance, tag string) error {
+func checkExistingImages(req router.Request, app *v1.AppInstance, tag string) (bool, error) {
 	images := &v1.ImageInstanceList{}
 	err := req.List(images, &kclient.ListOptions{
 		Namespace: app.Namespace,
 	})
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	for _, image := range images.Items {
-		if len(image.Tags) == 0 {
-			continue
-		}
-
-		newTags := make([]string, 0, len(image.Tags))
 		for _, existingTag := range image.Tags {
```

Comment:
I am sure you tested, but how does this logic compare the repo? Is that listing all images and only compares on tags?

Reply:
I think my response to your other question also addresses this.